### DISCUSSION TITLE
How to set the type with combine?

### DISCUSSION THREAD

**Discussion by kaisiant** [pmndrs/zustand#219] [Q&A] [answered]
Hello, I want to set the **cartItems** with the type but I don't know how to do it with combine.

I am using this way to set the type:
`cartItems: [] as Product[]`

Is there any other way to set it?

Full code:
```
export const useCartStore = create(
  combine(
    {
      showCartDrawer: false,
      cartItems: [] as Product[],
    },
    (set) => ({
      toggleCartDrawer: () =>
        set((state) => ({
          showCartDrawer: !state.showCartDrawer,
        })),
      addItemToCart: (item: Product) =>
        set((state) => ({
          cartItems: [...state.cartItems, item],
        })),
    })
  )
)
```

**Comment by dai-shi** [comment, marked as answer]
I suppose `as Product[]` is fine.

You could do `combine<PrimaryState, SecondaryState>()`, but then there's less reasoning to use combine. https://codesandbox.io/s/busy-dewdney-felo9?file=/src/App.tsx

**Reply by kaisiant** [reply]
Yup, I will just remove combine in this case. Thanks for the reply tho
